medienhaus/medienhaus-spaces
# Add retry handler for matrix rate limits

Uses `retry_after_ms` from the error response to await ratelimiting and retry the failed request.

- [x] fix bug where creating rooms loops and doesn't proceed with adding the room to its parent
- [x] display user feedback while syncing
- [x] use retry handler when creating application folder
- [x] use retry handler when creating service folders
- [x] use retry handler for /etherpad

**Testing:**

- try to encounter rate limits. for example create at least 8 sketches directly in spacedeck (not from within spaces! the more the better, since this should trigger ratelimiting by [matrix])
- open /spaces and go to /spacedeck
- /spacedeck will now try to sync all sketches and, if it runs into ratelimits, should automatically wait until it tries again

This PR continues from #71 after some technical difficulties 🧑‍🔧
